chia/wallet/wallet_state_manager.py

```diff
@@ -654,10 +676,93 @@ async def determine_coin_type(
         if did_curried_args is not None:
             return await self.handle_did(did_curried_args, parent_coin_state, coin_state, coin_spend, peer)
 
+        # Check if the coin is clawback
+        clawback_args = match_clawback_puzzle(puzzle, solution, self.constants.MAX_BLOCK_COST_CLVM)
+        if clawback_args is not None:
+            return await self.handle_clawback(clawback_args, coin_state, coin_spend)
         await self.notification_manager.potentially_add_new_notification(coin_state, coin_spend)
 
         return None
 
+    async def update_merkle_coin(self, peer: WSChiaConnection, coin_state: CoinState):
+        # Get unspent merkle coin
+        unspent_coins = await self.merkle_coin_store.get_all_unspent_coins()
```

Comment:
It's doing two things.
1. Update the spent Merkle coins.
2. Auto spending some Clawback coins.
We need to get all unspent Clawback coins and check them one by one to see if we can spend them in a batch. This is the major reason why we need to get all unspent coins.

Reply:
Ideally, we need a timer to run auto spending. But I cannot find a good place to add this. If you know, please let me know.

Reply:
> It's doing two things.
> 
> 1. Update the spent Merkle coins.

This is what i meant will not longer required if we just treat the merkle coins as normal coins with extra mapped data somewhere. 

> 2. Auto spending some Clawback coins.
>    We need to get all unspent Clawback coins and check them one by one to see if we can spend them in a batch. This is the major reason why we need to get all unspent coins.

Yes i got that, but we don't need to get all of them for every single coin state we process which is the concern i wanted to raise here. It's especially concerning during sync where we would check them many times for no reason. Seems like it would be better to trigger it if we are synced and a new peak was processed or so?